libcaf_test/caf/test/fixture/deterministic.hpp

```diff
@@ -502,6 +502,14 @@ class CAF_TEST_EXPORT deterministic {
   /// Dispatches all pending messages.
   size_t dispatch_messages();
 
+  /// Dispatches all pending messages.
+  template <class T = caf::resumable>
+  T& next_message() {
+    if (events_.empty())
+      CAF_RAISE_ERROR("events.empty()");
+    return dynamic_cast<T&>(*(events_.front()->target));
+  }
+
```

Comment:
This makes no sense to me. The comment seems copied from `dispatch_messages`, the name suggests it would return a message, but it actually seems to pull a reference to the next scheduled actor?

It seems like this is only being used in `request_timeout.test.cpp`. Looking at the test, we can just drop this function and all of the checks based on it without loosing much. There's one place where we are checking state:

```cpp
    check(next_message<ping_actor>().state().had_first_timeout);
```

I think that should be just:

```cpp
    auto testee_ptr = actor_cast<abstract_actor*>(testee);
    check(dynamic_cast<ping_actor&>(*testee_ptr).state().had_first_timeout);
```